extension/oidcauthextension/extension.go

```diff
@@ -167,7 +184,7 @@ func getGroupsFromClaims(claims map[string]any, groupsClaim string) ([]string, e
 	return []string{}, nil
 }
 
-func getProviderForConfig(config *Config) (*oidc.Provider, error) {
+func getProviderForConfig(ctx context.Context, config *Config) (*oidc.Provider, *http.Client, *http.Transport, error) {
```

Comment:
I have a hard time seeing the need for four return values here, is there way way we can consolidate functionality so we don't have to keep track of so many moving parts here?

Reply:
I can create a new struct for that. What do you think?

Reply:
Why not just make this a method on the `oidcExtension` and rename it to `setProviderForConfig` that only returns `err`. Then it could just set `e.transport`, `e.client`, and `e.provider` when necessary. It just seems like we need to consolidate the logic in one place or the other, rather than passing it all around.

Reply:
It sounds great. I'll try. Thanks for your feedback!